sys/include/net/netopt.h

```diff
@@ -269,6 +269,54 @@ typedef enum {
      */
     NETOPT_LAST_ED_LEVEL,
 
+    /** @brief   Radio modulation bandwidth
+     *
+     * Depends on the type of radio used and the regional regulatory and is
+     * used with LoRa/FSK modems.
+     *
+     * Actual implementations support 125kHz, 250kHz and 500kHz.
+     *
+     * @see drivers_sx127x
+     */
+    NETOPT_BANDWIDTH,
+
+    /** @brief   Radio spreading factor
+     *
+     * For LoRa/FSK modems, allowed values are in the range 7..12.
+     */
+    NETOPT_SPREADING_FACTOR,
+
+    /** @brief   Radio coding rate
+     *
+     * For LoRa/FSK modems, allowed values are in the range 5..8.
+     */
+    NETOPT_CODING_RATE,
+
+    /** @brief   Semtech module modem type.
```

Comment:
I think it will be useful to be able to switch from one modem to other using ifconfig. This is not possible right now but it will be possible with loramac adaption.